lib/cartopy/img_transform.py

```diff
@@ -358,7 +358,12 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
         if np.ma.isMaskedArray(new_array):
             new_array[non_self_inverse_points] = np.ma.masked
         else:
-            new_array = np.ma.array(new_array, mask=non_self_inverse_points)
+            new_array = np.ma.array(new_array, mask=False)
+            if new_array.ndim == 3:
+                for i in range(new_array.shape[2]):
+                    new_array[non_self_inverse_points, i] = np.ma.masked
```

Comment:
> @pelson Unfortunately, I don't have a test case. As I have a job now, it is unlikely I will find time to construct it.

That is quite understandable. Sounds like we're looking to transform an 3 channel image onto a non "verticalish" projection?